src/fatal.c

```diff
@@ -62,15 +62,15 @@ void fatal_log_message(const char* tag, const char* message, va_list args) {
 
     log_message(tag, LOG_LEVEL_ERROR, "A fatal error has been throw, unable to continue.");
     log_message(tag, LOG_LEVEL_ERROR, "Please, review the details below.");
-    log_message_internal(tag, LOG_LEVEL_ERROR, message, args);
+    log_message(tag, LOG_LEVEL_ERROR, message, args);
```

Comment:
args is a `va_list` and can't be processed by log_message that instead expects a list of arguments, log_message_internal is instead accepting the `va_list`. You had to switch to use `log_message` because you can't pass the output within `fatal_log_message`.

I think there are 2 options here:
- you can implement a `log_message_vargs` (or something with a similar name) that will implement the log_message logic but will work directly with a `va_list`
- you can move the logic that actually prints out the text (the various fprintf) in the sink using a function pointer, basically when you create the sink you pass a function pointer that gets the stuff that has to be printed out and to the hard job because, in this way, you will not need to pass the file pointer around

The first option is the quick-est one the second is the most correct one, personally at this stage I think that the most correct one can be implemented at a later stage, it's really up to you, both works for me.

Reply:
I just ran the main (`cachegrand target`) and it prints out

```
[2020-05-24T19:19:57Z][ERROR      ][main] KAbooommmmm -5616
```

instead of
```
[2020-05-24T19:19:57Z][ERROR      ][main] KAbooommmmm 0
```

because it's printing out (I am assuming) the pointer to `va_list` memory.